src/firefox/ffBrowser.ts

```diff
@@ -291,10 +291,18 @@ export class FFBrowserContext extends BrowserContextBase {
     await this._browser._connection.send('Browser.addBinding', { browserContextId: this._browserContextId || undefined, name, script: binding.source });
   }
 
-  async route(url: types.URLMatch, handler: network.RouteHandler): Promise<void> {
-    this._routes.push({ url, handler });
+  async route(url: types.URLMatch, handler: network.RouteHandler): Promise<() => Promise<void>> {
```

Comment:
A lot of repetition here across browsers, consider moving to BrowserContextBase.